spec/requests/cloud_volumes_spec.rb

```diff
@@ -310,4 +310,20 @@
       expect(response).to have_http_status(:ok)
     end
   end
+
+  describe 'create snapshot' do
+    it 'it can create cloud volume snapshot through POST' do
+      zone = FactoryBot.create(:zone)
+      provider = FactoryBot.create(:ems_autosde, :zone => zone)
+      cloud_volume = FactoryBot.create(:cloud_volume_autosde, :ext_management_system => provider)
+
+      api_basic_authorize(action_identifier(:cloud_volumes, :create_snapshot, :resource_actions, :post))
+      stub_supports(cloud_volume.class, :snapshot_create)
+
+      payload = {:action => "create_snapshot", :resources => {:snapshot_name => "stub_snapshot_name"}}
+      post(api_cloud_volume_url(nil, cloud_volume), :params => payload)
+
+      expect(response).to have_http_status(:ok)
```

Comment:
If it works, please use `expect_single_action_result` with something like `:success => true, :message => /Creating/, :task => true`. You can probably find a lot of examples of this method.

Just saying "it didn't return an error" will catch fewer errors than expecting a certain message back.